doc/doxygen/src/driver-guide.md

```diff
@@ -0,0 +1,381 @@
+Writing a Device Driver in RIOT                                 {#driver-guide}
+===============================
+
+This document describes the requirement, design objectives, and some
+non-function details when writing device drivers in/for RIOT. The term device
+driver in this context includes all 'CPU-external' devices connected to the CPU
+typically via peripherals like SPI, I2C, UART, GPIO, and similar. CPU
+peripherals itself are in RIOT not considered to be device drivers, but
+peripheral or low-level drivers. Typical devices are network devices like
+radios, Ethernet adapters, sensors, and actuators.
+
+[TOC]
+
+# General Design Objectives                   {#driver-guide-design-objectives}
+
+Device drivers should be as easy to use as possible. This implies an
+'intialize->ready' paradigm, meaning, that device drivers should be ready to use
+and in operation right after they have been initialized. On top, devices should
+work with physical values where ever possible. So e.g. sensors should return
```

Comment:
```suggestion
work with physical values wherever possible. So e.g. sensors should return
```

Reply:
I applied the typo fix ("where ever" --> "wherever"). I agree with the reformulation, but I don't want to introduce more changes in this PR than fixing typos and grammar mistakes, so that the discussion is focused only on whether we want this doc to remain in the wiki, or move it to the official doc.